via/via_system_macos.cpp

```diff
@@ -0,0 +1,1431 @@
+    std::ifstream *stream = NULL;
+    Json::Value root = Json::nullValue;
+    Json::Value inst_exts = Json::nullValue;
+    Json::Value dev_exts = Json::nullValue;
+    Json::Reader reader;
+    std::string full_driver_path;
+    char generic_string[2048];
+    uint32_t j = 0;
+
+    stream = new std::ifstream(cur_driver_json.c_str(), std::ifstream::in);
+    if (nullptr == stream || stream->fail()) {
+        PrintBeginTableRow();
+        PrintTableElement("");
+        PrintTableElement("Error reading JSON file");
+        PrintTableElement(cur_driver_json);
+        PrintEndTableRow();
+        goto out;
+    }
+
+    if (!reader.parse(*stream, root, false) || root.isNull()) {
+        PrintBeginTableRow();
+        PrintTableElement("");
+        PrintTableElement("Error reading JSON file");
+        PrintTableElement(reader.getFormattedErrorMessages());
+        PrintEndTableRow();
+        goto out;
+    }
+
+    PrintBeginTableRow();
+    PrintTableElement("");
+    PrintTableElement("JSON File Version");
+    if (!root["file_format_version"].isNull()) {
+        PrintTableElement(root["file_format_version"].asString());
+    } else {
+        PrintTableElement("MISSING!");
+    }
+    PrintEndTableRow();
+
+    if (root["ICD"].isNull()) {
+        PrintBeginTableRow();
+        PrintTableElement("");
+        PrintTableElement("ICD Section");
+        PrintTableElement("MISSING!");
+        PrintEndTableRow();
+        goto out;
+    }
+
+    found_json = true;
+
+    PrintBeginTableRow();
+    PrintTableElement("");
+    PrintTableElement("API Version");
+    if (!root["ICD"]["api_version"].isNull()) {
+        PrintTableElement(root["ICD"]["api_version"].asString());
+    } else {
+        PrintTableElement("MISSING!");
+    }
+    PrintEndTableRow();
+
+    PrintBeginTableRow();
+    PrintTableElement("");
+    PrintTableElement("Library Path");
+    if (!root["ICD"]["library_path"].isNull()) {
+        std::string driver_name = root["ICD"]["library_path"].asString();
+        std::string location;
+        bool could_load = true;
+        std::string load_error;
+        PrintTableElement(driver_name);
+        PrintEndTableRow();
+
+        if (DetermineJsonLibraryPath(cur_driver_json.c_str(), driver_name.c_str(), full_driver_path)) {
+            // First try the generated path.
+            if (access(full_driver_path.c_str(), R_OK) != -1) {
+                found_lib = true;
+                could_load = VerifyOpen(full_driver_path, load_error);
+            } else if (driver_name.find("/") == std::string::npos) {
+                if (FindMacOSSystemObject(this, driver_name, location, CheckDriver, true)) {
+                    found_lib = true;
+                    could_load = VerifyOpen(location, load_error);
+                }
+            }
+        }
+        if (!found_lib) {
+            uint32_t image_count = _dyld_image_count();
+            if (image_count == 0) {
+                    snprintf(generic_string, 1023,
```

Comment:
I don't think you need this check - if you just have the for-loop on image_count and then the if(!found) afterwards, it will catch the case where image_count is 0 and then there's only one copy of this HTML output.  Do you have time to make this edit?